flink-core/src/main/java/org/apache/flink/configuration/JobManagerOptions.java

```diff
@@ -106,17 +106,21 @@
 	/**
 	 * This option specifies the failover strategy, i.e. how the job computation recovers from task failures.
 	 */
-	@Documentation.ExcludeFromDocumentation("The failover strategy feature is highly experimental.")
 	public static final ConfigOption<String> EXECUTION_FAILOVER_STRATEGY =
 		key("jobmanager.execution.failover-strategy")
 			.defaultValue("full")
 			.withDescription(Description.builder()
-				.text("This option specifies how the job computation recovers from task failures. " +
+				.text("This option specifies the failover strategy, i.e. " +
+					"how the job computation recovers from task failures. " +
 					"Accepted values are:")
 				.list(
-					text("'full': Restarts all tasks."),
-					text("'individual': Restarts only the failed task. Should only be used if all tasks are independent components."),
-					text("'region': Restarts all tasks that could be affected by the task failure.")
+					text("'full': Use RestartAllStrategy to restarts all tasks to recover the job."),
+					text("'region': Use AdaptedRestartPipelinedRegionStrategyNG to restart all tasks " +
```

Comment:
I would avoid using class names, such as `AdaptedRestartPipelinedRegionStrategyNG`, here. We should describe their behavior only. 

Another idea that @zentol had, is to extend the section about restart strategies [1] with more detailed information about the failover strategies (`full` and `region`). 

[1] https://ci.apache.org/projects/flink/flink-docs-master/dev/restart_strategies.html

Reply:
Ok. Let's remove the class name and just describe the behavior here.

I think adding more information for different failover strategies is a very good idea. In this PR I'm trying to make the documentation easy to understand, but seems it's a bit hard to make it very clear if we do not want to add too many words for the option documentation.
Just not sure if the RestartStrategy page is the proper place? As the failover strategy and restart strategy are somehow independent.

Reply:
From a users perspective restart and failover strategies are connected quite closely, with one defining the "how" and the other defining "how often".
From a technical perspective they also aren't independent, since if restarts are disabled the failover strategy will not kick in.

Reply:
Just feel that adding a "Failover Strategies" section in "Restart Strategies" might be a bit confusing to users, indicating that failover strategy is part of restart strategy.  

Would it be better to re-org the page as below?
**Task Failure Recovery**
  +- Restart Strategies
     +- Fixed Delay Restart Strategy
     +- Failure Rate Restart Strategy
     +- ...
  +- Failover Strategies
     +- Restart All Strategy
     +- Restart Pipelined Region Strategy

Reply:
I would be ok with changing the heading of the page. If you are changing the file name, I'd introduce a redirect so that we do not break existing hyperlinks (see https://github.com/apache/flink/tree/9cdae7307e9be42be473a22d5a3e924c159ca12d/docs/redirects)

Reply:
Thanks for the reminding. Will take it this way.